.github/workflows/router-e2e.yml

```diff
@@ -0,0 +1,71 @@
+name: Router Maestro E2E
+
+on:
+  pull_request:
+    paths:
+      - packages/@expo/cli/**
+      - packages/expo-router/**
+
+jobs:
+  router-maestro:
+    runs-on: macos-14
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Use Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: '20.x'
+
+      - name: Install dependencies
+        run: YARN_IGNORE_SCRIPTS=true yarn --frozen-lockfile
```

Comment:
Might be a good idea to cache node modules. You can do this easily by using our own composite action: https://github.com/expo/expo/blob/main/.github/actions/expo-caches/action.yml

Reply:
How is the Expo Action different from enabling caching with [`actions/setup-node@v4`](https://github.com/actions/setup-node/blob/main/docs/advanced-usage.md#caching-packages-data)?